Dialogs/EventArgumentsDialog.cpp

```diff
@@ -0,0 +1,108 @@
+#include "EventArgumentsDialog.h"
+#include "Components/QMenuView.h"
+#include "MainWindow.h"
+#include "Models/TreeSortFilterProxyModel.h"
+
+#include <QComboBox>
+#include <QDialogButtonBox>
+#include <QGridLayout>
+#include <QLabel>
+#include <QLineEdit>
+#include <QMetaProperty>
+#include <QPushButton>
+#include <QSpinBox>
+#include <QToolButton>
+
+EventArgumentsDialog::EventArgumentsDialog(QWidget *parent, const QStringList &arguments) : QDialog(parent) {
```

Comment:
Why are the arguments string and not QVariant or something. Why aren't we using real reflection over the event proto here.

Reply:
Because they're strings... Arguments are "object", "integer" or "string"

Reply:
That's what's in event data? Strings is what the shared event stuff gives you to work with? I don't know why Josh would do it that way and not give an enum or some other typed way of doing it.